### PR TITLE
Update Grafana Agent install, config, and start topics

### PR DESCRIPTION
**Pull request by clayton-cornell**
#### PR Description

This PR is a followup to the install topic reorg PR ( https://github.com/grafana/agent/pull/4232 )

Changes
1. Update all references to Grafana Agent Flow
2. Add Next steps linking
3. Fix typos
4. Update note syntax
5. Add new Kubernetes config topic and move initial config content into new topic
6. Rework Other systems topic and rename to Standalone binary
7. Fix Optional step style

#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: docs/sources/flow/_index.md [hunk 1]
```diff
@@ -59,11 +59,11 @@ prometheus.remote_write "default" {
...
-* Learn about the core [Concepts][] of Grafana Agent Flow.
+* [Install][] Grafana Agent in flow mode.
+* Learn about the core [Concepts][] of Grafana Agent.
 * Follow our Grafana Agent Flow [Getting started][] guides.
-* Follow our [Tutorials][] to get started with Grafana Agent Flow.
-* Learn how to use Grafana Agent Flow's [Configuration language][].
+* Follow our [Tutorials][] to get started with Grafana Agent.
+* Learn how to use Grafana Agent's flow mode [Configuration language][].
 * Check out our [Reference][] documentation to find specific information you
   might be looking for.
```
Comment (rfratto) [inline docs/sources/flow/_index.md:68]: All of these are next steps for Flow mode, should we change them here to clarify?

for example, should "Learn about the core Concepts of Grafana Agent" say "Learn about the core Concepts of flow mode?"

File: docs/sources/flow/setup/configure/configure-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+## Kustomize considerations
+
+If using [Kustomize][] to inflate and install the [Helm chart][], be careful
+when using a `configMapGenerator` to generate the ConfigMap containing the
+configuration. By default, the generator appends a hash to the name and patches
+the resource mentioning it, triggering a rolling update.
+
+In the case of Grafana Agent, this behavior is undesirable, as the startup
+time can be significant depending on the size of the Write-Ahead Log. Instead,
+the [Helm chart][] provides a sidecar container that will watch the ConfigMap
```
Comment (rfratto) [inline docs/sources/flow/setup/configure/configure-kubernetes.md:46]: a link for Helm chart is missing here

File: docs/sources/flow/setup/configure/configure-linux.md [hunk 3]
```diff
@@ -1,13 +1,13 @@
 ---
-description: Learn how to configure Grafana Agent Flow on Linux
-title: Configure Grafana Agent Flow on Linux
+description: Learn how to configure Grafana Agent on Linux
```
Comment (rfratto) [inline docs/sources/flow/setup/configure/configure-linux.md:2]: It seems like some of these pages are losing context about what mode they apply to. Is that intentional? If you're linked to this page directly, the only way to identify that it's for Flow mode is to check the navbar and notice that it's in the flow mode section. Maybe that's ok? 

It looks like it's just the configure folder which doesn't clearly specify flow mode in the title/descriptions anymore.

File: docs/sources/flow/setup/install/docker.md [hunk 4]
```diff
@@ -21,8 +21,7 @@ Grafana Agent Flow is available as Docker images on the following platforms:
 
 * Ensure that [Docker][] is installed and running on your machine.
 
-* Ensure that you have an existing Grafana Agent Flow configuration file to
-  use saved on your host system, such as:
+* Ensure that you have an existing Grafana Agent river configuration file on your host system, such as:
```
Comment (rfratto) [inline docs/sources/flow/setup/install/docker.md:24]: River should be a proper noun a la YAML or JSON:

```suggestion
* Ensure that you have an existing Grafana Agent River configuration file on your host system, such as:
```

**Review by rfratto — APPROVED** [review]
LGTM with little bits of feedback

File: docs/sources/flow/setup/configure/configure-kubernetes.md (hunk 2 above)
Comment (clayton-cornell) [reply docs/sources/flow/setup/configure/configure-kubernetes.md:46]: Added it back in

File: docs/sources/flow/setup/install/docker.md (hunk 4 above)
Comment (clayton-cornell) [reply docs/sources/flow/setup/install/docker.md]: Fixed the typo

File: docs/sources/flow/_index.md (hunk 1 above)
Comment (clayton-cornell) [reply docs/sources/flow/_index.md]: Yes, there's a lot of room for improving the link text. I've updated this part a little.

File: docs/sources/flow/setup/configure/configure-linux.md (hunk 3 above)
Comment (clayton-cornell) [reply docs/sources/flow/setup/configure/configure-linux.md]: You're too fast on reviews :-) 

I've adjusted this text in the metadata and in the opening sentences. It's a balance between overloading the use of "flow'  and losing context s you've noted. I've been re-reading the content and adding flow where the context is weak.

**Review by rfratto — APPROVED** [review]
Looks good, thanks!

**Pull request merged**